src/Turing.jl

```diff
@@ -107,23 +112,22 @@ The dispatch is based on the value of a `sampler` variable.
 To include a new inference algorithm implements the requirements mentioned above in a separate file,
 then include that file at the end of this one.
 """
-mutable struct Sampler{T} <: AbstractSampler
+mutable struct Sampler{T, S<:SamplerState} <: AbstractSampler
     alg      ::  T
     info     ::  Dict{Symbol, Any} # sampler infomation
     selector ::  Selector
+    state    ::  S
 end
 Sampler(alg) = Sampler(alg, Selector())
 Sampler(alg, model::Model) = Sampler(alg, model, Selector())
-Sampler(alg, model::Model, s::Selector) = Sampler(alg, s)
+# Sampler(alg, model::Model, s::Selector) = Sampler(alg, model, s)
```

Comment:
Any reason for commenting out this constructor?

Reply:
Nope, I think that was left over from when I was fixing DyanmicHMC. Thanks for catching this!